### ISSUE TITLE
Adobe ColdFusion Compatibility?  Or only CommandBox?

### ISSUE THREAD

**Issue by JamoCA** [Ortus-Solutions/globber#1]
Does this work with Adobe ColdFusion?  If so, which version?

I attempted to use v3.0.7 with Adobe ColdFusion 2016 and there are errors due to incompatible syntax.  (I now that CF2016 is deprecated.  The last time I checked, it was still more popular than CF2021 and I am still in the process of testing & finding CF2021 bugs.)  I downgraded to v3.0.6 and determined that `setBaseDir` appears to be a CommandBox command.

If this is CommandBox, Lucee or version dependent, could the minimum requirements and any dependencies be listed?  Thanks.

**Comment by bdw429s** [comment]
At one time this worked on ACF and it is intended to do so. I did a bunch of refactoring a while back and I haven't tested it on ACF since.  My use of the elvis operator will likely make it difficult for it to run on 2016 due to the large amount of bugs there, but you're welcome to try.  As far as what's not working, you'll need to be more specific.  

>  determined that setBaseDir appears to be a CommandBox command.

I don't understand what you mean by this.  Assuming you're talking about the `Globber.cfc`, the code has this:

```js

component accessors="true" {
        
	/** Directory the list was pulled from */
	property name='baseDir' default='';
```
so the `setBaseDir()` method is just an auto-generated setter method from CF.

**Comment by JamoCA** [comment]
ah, you are right regarding `setBaseDir`. I searched the code and forgot to check accessors. (I searched online and saw something related to CommandBox and thought it may be a built-in function.)

I'll test and see if I can dumb it down so that it is compatible with ACF 2016.

**Comment by bdw429s** [comment]
Yes, please document any issues you find.  I'd love to have compat, but I haven't put forth much effort on it myself since my immediate needs for this have been within CommandBox/Lucee.  There is a full unit test suite you can run to test every feature of the library.  You should be able to just `box install` and `box server start` and then navigate to `/tests/runner.cfm` to run the test suite locally.

**Comment by JamoCA** [comment]
I can't even initialize the CFC using ColdFusion 2016. I believe it's due to the use of closures going back as far as Mar 22, 2017 "Introduce new Globber fluent API".  (ie, for DirectoryList, Adobe didn't even add named parameters until CF2018. It supports a filter function, but probably not a closure.)

The [State of the CF Union 2021 Survey](https://teratech.com/state-of-the-cf-union-2021-results) revealed that a large number of users were still using CF2016 (at the time of survey, it more than doubled CF2021 usage).  I haven't upgraded yet (due to Adobe bugs) and was hoping that this library would be beneficial during the migration.
